graph-gateway/src/network/service.rs

```diff
@@ -87,17 +88,16 @@ impl ResolvedSubgraphInfo {
 /// To create a new [`NetworkService`] instance, use the [`NetworkServiceBuilder`].
 #[derive(Clone)]
 pub struct NetworkService {
-    network: Eventual<Ptr<NetworkTopologySnapshot>>,
+    pub network: watch::Receiver<NetworkTopologySnapshot>,
```

Comment:
Let's keep the fields private. This will help us evolve the service without changing too many things. (I believe it is the _Open-closed_ principle of the SOLID principles.)

I think we can avoid accessing this field from outside, for example, by providing a method that returns a watch channel returning a derived data structure.

Reply:
I didn't think about that possibility. Makes sense to me 👍🏼